### PR TITLE
Chat flow bug fixing

### PR DESCRIPTION
**Pull request by alexanishchenko88**
#### 📲 What

Chat Flow Bug fixing allowing chat skip and updating the answers structure from firebase.

#### 🤔 Why
		
Bug fixing to ensure the correct chat flow
		
#### 🛠 How
		
Updated the data we are storing into firebase after chat configuration finishes. Now instead of simple map we will have Map<String, Map<String, String>>

Added:
ChatResponseKeys.dart

Updated:
All the files related with Map structure update

Deleted:
-

#### ✅ Acceptance criteria

- [ ] Design Review for UI with BA completed. 
- [ ] Manually tested and verified on Android device/emulator (min. Lollipop 5.1)
- [ ] Showcase video / automation test
- [ ] Passing all tests
- [ ] Rebased/merged with latest changes from development and re-tested
- [ ] Removed unsed comments. TODOs must have JIRA for future work.

#### Coding Standards Checklist
- [ ] unit tests 80% coverage on testable code (functions, methods, class)
- [ ] (Architecture) redux state, reducers, actions, middleware defined under lib/redux
- [ ] ui elements defined under lib/ui. 
- [ ] Strings ready for localisation (e.g. defined in lib/utils/strings.dart)
- [ ] Assets (images/icons path) defined in lib/utils/assets.dart
- [ ] Colors defined in lib/utils/colors.dart
- [ ] Dimensions (ie margins padding) defined in lib/utils/dimens.dart
- [ ] TextStyles defined in lib/utils/styles.dart

Recommended Style Guide: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo

#### 🕵️‍♂️ How to test

Notes for QA

#### Reviewers

@farmsmart/amido @farmsmart/wamf

### CODE REVIEW

File: lib/chat/utils/ChatResponseKeys.dart [hunk 1]
```diff
@@ -0,0 +1,6 @@
+//TODO not sure about this class, if should be here or if even should exist.
+class ChatResponseKeys {
+  static const keyId = "id";
+  static const keyTitle = "title";
+  static const keyValue = "value";
+}
```
Comment (alexanishchenko88) [inline lib/chat/utils/ChatResponseKeys.dart]: As commented right above, not sure about this class, if we should persist that or maybe move it into some other place.

Comment (LeeMatthewHiggins) [reply lib/chat/utils/ChatResponseKeys.dart]: we normally would make it private at the point of use, and call it "_Fields"

Comment (LeeMatthewHiggins) [reply lib/chat/utils/ChatResponseKeys.dart]: we don't need to repeat the context "key"

Comment (alexanishchenko88) [reply lib/chat/utils/ChatResponseKeys.dart]: Ok, changed, moved to private _Fields class at the point of usage, ChatResponseKeys.dart has been removed, **key** annotation at the beginning of each property removed.

Comment (LeeMatthewHiggins) [reply lib/chat/utils/ChatResponseKeys.dart]: 👍

**Review by LeeMatthewHiggins — COMMENTED** [review]
is this still WIP or ready to be reviewed?

**Review by LeeMatthewHiggins — APPROVED** [review]
Updated and rebased

**Pull request merged**